src/atoms/Text/Text.web.js

```diff
@@ -1,3 +1,117 @@
-const Text = () => null;
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import baseTheme from '../../tokens/theme';
+import { getColorKeys, getColor, getLineHeight } from '../../_helpers/theme';
+
+const styles = {
+  fontFamily({ theme, _weight }) {
+    return theme.fonts.family.lato[_weight];
+  },
+  fontSize({ theme, size }) {
+    return theme.fonts.size[size];
+  },
+  color({ theme, color }) {
+    return getColor(theme, color);
+  },
+  textDecorationLine({ _isUnderlined }) {
+    if (_isUnderlined) {
+      return 'underline';
+    } else {
+      return 'none';
+    }
+  },
+  align({ align }) {
+    switch (align) {
+      case 'left':
+      default:
+        return 'flex-start';
+      case 'center':
+        return 'center';
+      case 'right':
+        return 'flex-end';
+    }
+  },
+  letterSpacing({ theme, _letterSpacing }) {
+    return `${theme.fonts.letterSpacing[_letterSpacing]}`;
+  },
+  lineHeight({ theme, size, _lineHeight }) {
+    if (_lineHeight) {
+      return theme.fonts.lineHeight[_lineHeight];
+    } else {
+      return getLineHeight(theme, size);
+    }
+  },
+  overflow({ truncate }) {
+    if (truncate) {
+      return 'hidden';
+    } else {
+      return '';
+    }
+  },
+  whiteSpace({ truncate }) {
+    if (truncate) {
+      return 'nowrap';
+    } else {
+      return '';
+    }
+  },
+  textOverflow({ truncate }) {
+    if (truncate) {
+      return 'ellipsis';
+    } else {
+      return '';
+    }
+  },
+};
+
+const Text = styled(
+  ({
+    as,
+    fontFamily,
+    fontSize,
+    children,
+    textDecorationLine,
+    align,
+    letterSpacing,
+    lineHeight,
+    ...props
+  }) => React.createElement(as, props, children),
+)`
+  font-family: ${styles.fontFamily};
+  font-size: ${styles.fontSize};
+  color: ${styles.color};
+  text-decoration-line: ${styles.textDecorationLine};
+  align-self: ${styles.align};
+  letter-spacing: ${styles.letterSpacing};
+  line-height: ${styles.lineHeight};
+  overflow: ${styles.overflow};
+  white-space: ${styles.whiteSpace};
+  text-overflow: ${styles.textOverflow};
+`;
+
+Text.propTypes = {
+  as: PropTypes.string,
+  size: PropTypes.oneOf(['xxsmall', 'xsmall', 'small', 'medium', 'large']),
+  color: PropTypes.oneOf(getColorKeys()),
+  align: PropTypes.oneOf(['left', 'center', 'right']),
+  'data-testid': PropTypes.string,
+  _weight: PropTypes.oneOf(Object.keys(baseTheme.fonts.weight)),
+  _isUnderlined: PropTypes.bool,
+  _letterSpacing: PropTypes.oneOf(Object.keys(baseTheme.fonts.letterSpacing)),
+  _lineHeight: PropTypes.oneOf(Object.keys(baseTheme.fonts.lineHeight)),
+  truncate: PropTypes.bool,
```

Comment:
Put this above internal props.

Reply:
What is this prop & why are we deviating from a common API for web & mobile? :(

Reply:
Oh, yes. This is `numberOfLines` in native. 😀

Reply:
@susheelachoudry can we use a common API for web & mobile here?

Reply:
I am not sure that we numberOfLines in web. @deorukhkarchaitanya

Reply:
And native doesn't have truncate. We need to decide on a common API that maps to truncate on web & numberOfLines on native

Reply:
```numberOfLines``` will restrict the text to the mentioned number of lines. Can we replicate that behaviour on web?

Reply:
we cant replicate numberOfLines, as far as i know. but we can achieve via giving height. @deorukhkarchaitanya

Reply:
Seems tricky. Continuing this conversation on slack.

Reply:
done